addons/popochiu/engine/interfaces/i_dialog.gd

```diff
@@ -87,7 +87,25 @@ func finish_dialog() -> void:
 func say_selected() -> void:
 	await C.player.say(selected_option.text)
 
-
+## Transforms any text to gibberish preserving bbcode tags
+func create_gibberish(input_string: String) -> String:
+	var output_text: String = ""
+	var bbcode: bool = false
+	var letters = ['a','e','i','o','u','y','b','c','d','f','g','h','j','k','l','m','n','p','q','r','s','t','v','w','x','z']
+	for chr in input_string:
+		if(chr == '['):
+			bbcode = true
+			
+		if (!bbcode):
+			if (chr != ' '):
+					output_text += letters[randi_range(0,letters.size()-1)]
+			else:
+				output_text += ' ' 
+		else:
+			output_text += chr
+		if(chr == ']'):
```

Comment:
I would move this up, at line 98.

Reply:
the reason why is like that is that if i moved that checking car if it's ']' and change become=false then it won't add it to the text :) I can later change to something else :)

```
for chr in input_string:
		if(chr == '['):
			bbcode = true
		elif(chr == ']'):
			output_text += chr
			bbcode = false
			continue
		
		if (!bbcode):
			if (chr != ' '):
					output_text += letters[randi_range(0,letters.size()-1)]
			else:
				output_text += ' ' 
		else:
			output_text += chr
```

I need to think how to better approach this - it was just a suggestion of a change to get better with coding add-on and with codebase :)

Reply:
I will suggest using a custom effect for **BBCode**. I did something similar [in this repo](https://github.com/Perro-Viejo/gossip_buey-prototype). We can have something like `[gibber]my text[/gibber]`. So the only thing you'll need to do is to wrap the input **String** in those tags.

https://github.com/Perro-Viejo/gossip_buey-prototype/blob/main/popochiu/graphic_interface/dialog_text/highlight_text_effect.gd

The code is not polished, but basically in this script I changed the characters by a random glyph to reveal only those that depend on the `Globals.dialog_listened_indexes` which are the indexes in the array of characters of said **String**.